types/stripe-v3/index.d.ts

```diff
@@ -165,6 +197,94 @@ declare namespace stripe {
         metadata: any;
         name?: string;
         tokenization_method?: tokenizationType;
+        three_d_secure?: 'required' | 'recommended' | 'optional' | 'not_supported';
+    }
+
+    interface RetrieveSourceOptions {
+        id: string;
+        client_secret: string;
+    }
+
+    // Container for all payment request related types
+    namespace paymentRequest {
+        interface StripePaymentRequestUpdateOptions {
+            currency: string;
+            total: {
+                amount: number;
+                label: string;
+                pending?: boolean;
+            };
+            displayItems?: Array<{
+                amount: number;
+                label: string;
+                pending?: boolean;
+            }>;
+            shippingOptions?: ShippingOption[];
+        }
+
+        interface StripePaymentRequestOptions extends StripePaymentRequestUpdateOptions {
+            country: string;
+            requestPayerName?: boolean;
+            requestPayerEmail?: boolean;
+            requestPayerPhone?: boolean;
+            requestShipping?: boolean;
+        }
+
+        interface UpdateDetails {
+            status: 'success' | 'fail' | 'invalid_shipping_address';
+            total?: {
+                amount: number;
+                label: string;
+                pending?: boolean;
+            };
+            displayItems?: Array<{
```

Comment:
You can use the same interface here -> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23563/files#diff-16413613d705a8c8bc02299c76d35068R217